### ISSUE TITLE
Fileupload - onBeforeUpload - formData object is not inspectable

### ISSUE THREAD

**Issue by Exerlol** [primefaces/primeng#1670]
**I'm submitting a ...** 

[x] bug report => Search github for a similar issue or PR before submitting
[x] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35

- Current behavior

First of all let me explain what i'm trying to achieve, i'm trying to make some sort of custom validation for uploaded file before i send it to server, to do so I have to access formData before I upload file. I was googling for a while and i found out that formData object is not inspectable therefor i cannot manipulate or inspect file inside of onBeforeUpload event, what i can do is send it and inspect it on server side, but i'd like to make a client side validation as well,  so since your not shipping your ts files, i made a workaround in compiled js code and managed to get it to work. 
So to summarize: formData inside of  onBeforeUpload event is always undefined.

It would also be lovely if u could expose your onFileSelect event - in my opinion this could be really beneficial.

By the way, thanks for your plugin and keep up the good work!

- Expected behavior

I expect that i can access and modify formData (ie. File) on client side inside onBeforeUpload method.

- Minimal reproduction of the problem with instructions

Its easy to reproduce this problem, just try to console log formData inside onBeforeUpload method.

- Please tell us about your environment:
Working with VS Code IDE

- Angular version: 2.0.X
Using Angular 2.2.1 - but this is no angular 2 related problem
- PrimeNG version: 2.0.X
Using latest PrimeNG version

**Comment by cagataycivici** [comment]
Recent versions provide formData in onBeforeUpload.

**Comment by tonyawad88** [comment]
@cagataycivici `onBeforeUpload` passes in an event of type FileBeforeUploadEvent which has form data but it is not readable.  Looking to get the name of the file. 
![Screenshot 2024-08-29 at 7 54 10 PM](https://github.com/user-attachments/assets/3e87ab58-dfa4-4efb-bb5f-85fc41541c69)
